### ISSUE TITLE
go_image + k8s_object on macOS with --cpu k8

### ISSUE THREAD

**Issue by mtsgrd** [bazelbuild/rules_k8s#103]
I'm trying to deploy a go_image (rules_docker) from my mac (High Sierra) to a Kubernetes cluster and am seeing the following unexpected behavior.

```
bazel run --cpu k8 //risk/jeopardy_ui:image -- --norun  # A go_image rule
...
Loaded image ID: c9f1edf07b2e


bazel run --cpu k8 //risk/jeopardy_ui:deployment.create  # A k8s_object rule
...
# jeopardy_ui id deployed -> 67bc10da402d


bazel run --cpu k8 //risk/jeopardy_ui:full_deploy.create  # A k8s_objects rule
...
# jeopardy_ui id deployed -> 26d62db52b54
```

Running the container output by the first command works as expected, the other two error out with the following:
```
standard_init_linux.go:185: exec user process caused "exec format error"
```

In other words, I get different digests when deploying a go_image with `k8s_object`, as well as when said rule is triggered by a `k8s_objects` collection.

Not sure what other details I could provide, but would be happy if someone was able to shed some light on this?

**Comment by mattmoor** [comment]
Is the digest different when the image is pushed via `docker_push`?

It's possible that the digest is different when you publish the image with the Docker client.

If the digest is changing between `docker_push` and `k8s_object` it is very likely a bug, and may have to do with `cfg = "host"` vs. `cfg = "target"`.

**Comment by mtsgrd** [comment]
`container_push` gives me `c9f1edf07b2e`, the expected image!

**Comment by mtsgrd** [comment]
It's bedtime in Stockholm so I don't have time to try this, but would adding `cfg = "host"` to the following line do the trick? https://github.com/bazelbuild/rules_k8s/blob/master/k8s/object.bzl#L211

**Comment by mattmoor** [comment]
IIUC we want `cfg = "target",`.  `"host"` is intended for building tools to run on the build machine, e.g. the resolver.  `"target"` is intended for building things for the target environment.

**Comment by mattmoor** [comment]
Are you on HEAD, because I think this [was fixed](https://github.com/bazelbuild/rules_k8s/blob/master/k8s/objects.bzl#L45).

**Comment by mtsgrd** [comment]
Ah yes, target was what I meant. And yes, you're right, it's fixed in head and works great. Also, I know you're paid to do this, but all of this code is so much appreciated. :)

**Comment by mattmoor** [comment]
LOL, I'm glad!  Please don't hesitate to reach out if you have any further problems.
